zeppelin-plugins/launcher/k8s-standard/src/main/java/org/apache/zeppelin/interpreter/launcher/K8sRemoteInterpreterProcess.java

```diff
@@ -403,28 +421,35 @@ boolean isSparkOnKubernetes(Properties interpreterProperties) {
   }
 
   @VisibleForTesting
-  String buildSparkSubmitOptions(String userName) {
-    StringBuilder options = new StringBuilder();
-
-    options.append(" --master k8s://https://kubernetes.default.svc");
-    options.append(" --deploy-mode client");
-    if (properties.containsKey(SPARK_DRIVER_MEMORY)) {
-      options.append(" --driver-memory ").append(properties.get(SPARK_DRIVER_MEMORY));
-    }
+  String prepareZeppelinSparkConf(String userName) {
+    StringJoiner sparkConfSJ = new StringJoiner("|");
```

Comment:
Sorry for the fundamental question, but I wonder if `|` is correct?

Reply:
Yes `|` is correct. Take a look into https://github.com/apache/zeppelin/blob/4e9cab538e75faa9f4168143f1fdd25c27abedb2/bin/interpreter.sh#L259
and the discussion https://github.com/apache/zeppelin/pull/4127.

Reply:
Aha ~ Thank you!